Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -133,34 +141,63 @@ internal void ValidateSecurePassword(SecureString password)
         private string ExtractPassword(SecureString password) =>
             password == null ? string.Empty : SecureStringHelper.Decode(password);
 
-        internal SFSession GetSession(string connStr, SecureString password)
+        internal SFSession GetSession(string connStr, SecureString password, SecureString passcode)
         {
             s_logger.Debug("SessionPool::GetSession" + PoolIdentification());
+            SFSession session = null;
```

Comment:
couldn't this happen after the check of `if (!GetPooling())...`?

Reply:
Done